feature/gribi/ate_tests/base_leader_election_test/base_leader_election_test.go

```diff
@@ -182,201 +180,68 @@ func testTraffic(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology,
-			WithIndex(nhIndex).
-			WithIPAddress("192.0.2.6").
-			WithElectionID(12, 0))
-
-	args.clientA.Modify().AddEntry(t,
-		fluent.NextHopGroupEntry().
-			WithNetworkInstance(instance).
-			WithID(nhgIndex).
-			AddNextHop(nhIndex, 1).
-			WithElectionID(12, 0))
-
-	args.clientA.Modify().AddEntry(t,
-		fluent.IPv4Entry().
-			WithPrefix(ateDstNetCIDR).
-			WithNetworkInstance(instance).
-			WithNextHopGroup(nhgIndex).
-			WithElectionID(12, 0))
-
-	if err := awaitTimeout(args.ctx, args.clientA, t, time.Minute); err != nil {
-		t.Fatalf("Could not program entries via clientA, got err: %v", err)
-	}
-
-	chk.HasResult(t, args.clientA.Results(t),
-		fluent.OperationResult().
-			WithNextHopOperation(nhIndex).
-			WithOperationType(constants.Add).
-			WithProgrammingResult(fluent.InstalledInRIB).
-			AsResult(),
-		chk.IgnoreOperationID(),
-	)
-
-	chk.HasResult(t, args.clientA.Results(t),
-		fluent.OperationResult().
-			WithNextHopGroupOperation(nhgIndex).
-			WithOperationType(constants.Add).
-			WithProgrammingResult(fluent.InstalledInRIB).
-			AsResult(),
-		chk.IgnoreOperationID(),
-	)
-
-	chk.HasResult(t, args.clientA.Results(t),
-		fluent.OperationResult().
-			WithIPv4Operation(ateDstNetCIDR).
-			WithOperationType(constants.Add).
-			WithProgrammingResult(fluent.InstalledInRIB).
-			AsResult(),
-		chk.IgnoreOperationID(),
-	)
+// configureIPv4ViaLeaderClientInstalled configures a IPv4 Entry via the leader Client.
+// It ensures that the entry is installed.
+func configureIPv4ViaLeaderClientInstalled(t *testing.T, args *testArgs, leader *gribi.GRIBIHandler, nh string) {
+	t.Logf("Adding an IPv4Entry for %s pointing to ATE port-2 via a client as leader.", ateDstNetCIDR)
+	leader.AddNH(t, nhIndex, nh, instance, fluent.InstalledInRIB)
+	leader.AddNHG(t, nhgIndex, map[uint64]uint64{nhIndex: 1}, instance, fluent.InstalledInRIB)
+	leader.AddIPv4(t, ateDstNetCIDR, nhgIndex, instance, "", fluent.InstalledInRIB)
 }
 
-// testIPv4LeaderActiveChange modifies election ID of ClientA with an Election ID of 12
-// and configures a IPv4 entry through this client. Ensure that the entry via ClientA
-// is active through AFT Telemetry.
+// testIPv4LeaderActiveChange first makes the cleintB leader.
+// Second it configures an IPV4 Entry through clinetB and ensures
+// that the entry is avtive by checking AFT Telemetry and traffic.
+// Thrid, it configures an IPv4 entry through clientA without
+// making it the leader and ensures that the installation fails.
+// Forth, it makes the ClientA master, configures an IPV4 through clinetA
+// and ensures that the entry is active by checking AFT Telemetry and traffic.
 func testIPv4LeaderActiveChange(ctx context.Context, t *testing.T, args *testArgs) {
-	// Configure IPv4 route for 198.51.100.0/24 pointing to ATE port-3 via clientB.
-	configureIPv4ViaClientB(t, args)
+	// Configure IPv4 route for 198.51.100.0/24 pointing to ATE port-3 via clientB as the leader.
+	args.clientB.BecomeLeader(t)
```

Comment:
Any assumption based on the election id can makes the test case flaky since the client can only be leader when it uses an election id higher than the last known election id (can be any number) on the server side, i.e., there is no guarantee that client can be master  by setting election id to 11. So, I suggest to modify the test case description accordingly by using the last know election id +1 (BecomeLeader does this) rather than a fixed number.

Reply:
I see what you're saying. I can see that the test can try to work around some of the problems about accumulating device states from previous runs, but ultimately I think our static binding and other Ondatra bindings should be able to consistently reset the device to a known state between test runs.

The reservation system that we use internally already does that, but this hasn't been implemented in the static binding. Most of the test plans were written assuming a clean state, and I think that's not only for clarity but also for a pragmatic reason. Some of the states are finite resources, such as available LAG IDs or VLAN IDs that run out quickly after repeated test runs.

@greg-dennis and I will be working something out about device state reset. I think for now, we can do BecomeLeader here, but let's mark it as a TODO in the test plan to revert back to hard-coded election IDs. Floating election IDs could be annoying to debug when a test is run on a non-compliant device.

Reply:
By the way, is there a way to reset gRIBI election ID on Cisco?

Reply:
There is no normal procedure to do that. We probably need to reset the gribi process. @mingyangcisco  please correct me if I am wrong?

Reply:
Continue discussion in #147